src/node_crypto.cc

```diff
@@ -745,6 +748,56 @@ void SecureContext::SetKey(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+#ifndef OPENSSL_NO_ENGINE
+// Helpers for the smart pointer.
+void ENGINE_free_fn(ENGINE* engine) { ENGINE_free(engine); }
+
+void ENGINE_finish_and_free_fn(ENGINE* engine) {
+  ENGINE_finish(engine);
+  ENGINE_free(engine);
+}
+
+void SecureContext::SetEngineKey(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  SecureContext* sc;
+  ASSIGN_OR_RETURN_UNWRAP(&sc, args.Holder());
+
+  CHECK_EQ(args.Length(), 2);
+
+  char errmsg[1024];
+  const node::Utf8Value engine_id(env->isolate(), args[1]);
+  std::unique_ptr<ENGINE, std::function<void(ENGINE*)>> e =
```

Comment:
Is adding into node_crypto.h a `DeleteFnPtr<ENGINE, ENGINE_finish_and_free_fn>` not possible? If it doesn't work, this is OK, I just want to make sure you noticed the existing smart ptr pattern.

Reply:
Well, there is a slight variation in that I first create a pointer with a `ENGINE_free_fn` as a deleter, and then replace it with `ENGINE_finish_and_free_fn`. I'm not sure if that will work with `DeleteFnPtr`.

Reply:
makes sense